src/main/java/uk/gov/justice/hmpps/prison/service/transformers/OffenderAlertTransformer.java

```diff
@@ -52,16 +67,16 @@ private static Alert transform(final OffenderAlert offenderAlert) {
         return Alert
             .builder()
             .alertId(offenderAlert.getSequence().longValue())
-            .alertCode(Optional.ofNullable(offenderAlert.getCode()).map(AlertCode::getCode).orElse(null))
-            .alertType(Optional.ofNullable(offenderAlert.getType()).map(AlertType::getCode).orElse(null))
+            .alertCode(offenderAlert.getAlertCode())
+            .alertType(offenderAlert.getAlertType())
             .alertCodeDescription(Optional
                 .ofNullable(offenderAlert.getCode())
                 .map(AlertCode::getDescription)
-                .orElse(null))
+                .orElse(offenderAlert.getAlertCode()))
```

Comment:
consistent with existing functionality, return code when description can not be found